plumbing/object/file_test.go

```diff
@@ -247,3 +247,29 @@ func (s *FileSuite) TestFileIter(c *C) {
 
 	c.Assert(count, Equals, 1)
 }
+
+func (s *FileSuite) TestFileIterSubmodule(c *C) {
+	st, err := filesystem.NewStorage(fixtures.ByTag("submodule").One().DotGit())
+	c.Assert(err, IsNil)
+
+	hash := plumbing.NewHash("a692ec699bff9117c1ed91752afbb7d9d272ebef")
+	commit, err := GetCommit(st, hash)
+	c.Assert(err, IsNil)
+
+	tree, err := commit.Tree()
+	c.Assert(err, IsNil)
+
+	expected := []string{
+		".gitmodules",
+	}
+
+	var count int
+	i := tree.Files()
+	i.ForEach(func(f *File) error {
```

Comment:
There are several things that are wrong with this test:

- it is too complex to test that the obtained must be a single file.
- If Files return an empty list of file, I'm pretty sure the test will pass, so we will never know if submodules are ignored.
- the name of the test does not explain what it is doing (check that submodules are ignored)
- the test is not calling the function that has the logic under test (FileIter.Next)